src/AppClassComponent.js

```diff
@@ -0,0 +1,65 @@
+import React, { Component } from 'react'
+
+class App extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            index: 1,
+            image: null,
+            allowUpdates: true
+        }
+    }
+
+    fetchImage() {
+        const url = `https://jsonplaceholder.typicode.com/photos/${this.state.index}`
+        return new Promise((resolve) => {
+            setTimeout(() => {
+                fetch(url)
+                    .then(resp => resp.json())
+                    .then(resp => resolve(resp))
+            }, 200)
+        })
+    }
+
+    componentDidMount() {
+        this.fetchImage().then(res => {
+            this.state.allowUpdates && this.setState(() => ({image: res}))
+        });
+    }
+
+    componentDidUpdate(prevProps, prevState, snapshot) {
+        if (this.state.index === prevState.index) return;
+        this.fetchImage().then(res => {
+            this.state.allowUpdates && this.setState(() => ({image: res}))
```

Comment:
@luisaugusto I think we would want to remove `this.state.allowUpdates`. it gives away an answer. Unless the idea is to make them redesign the App component from class to functional.